server/src/main/java/org/opensearch/extensions/action/ExtensionActions.java

```diff
@@ -0,0 +1,162 @@
+                    response.setResponseBytes(actionResponse.getResponseBytes());
+                    inProgressLatch.countDown();
+                }
+
+                @Override
+                public void onFailure(Exception exp) {
+                    logger.debug("Transport request failed", exp);
+                    byte[] responseBytes = ("Request failed: " + exp.getMessage()).getBytes(StandardCharsets.UTF_8);
+                    response.setResponseBytes(responseBytes);
+                    inProgressLatch.countDown();
+                }
+            }
+        );
+        inProgressLatch.await(5, TimeUnit.SECONDS);
+        return response;
+    }
+
+    public ExtensionActionResponse sendTransportRequestToExtension(ExtensionActionRequest request) throws InterruptedException {
+        DiscoveryExtension extension = actionsMap.get(request.getAction());
+        if (extension == null) {
+            throw new ActionNotFoundTransportException(request.getAction());
+        }
+        final CountDownLatch inProgressLatch = new CountDownLatch(1);
+        final ExtensionActionResponse extensionActionResponse = new ExtensionActionResponse(new byte[0]);
+        final TransportResponseHandler<ExtensionActionResponse> extensionActionResponseTransportResponseHandler =
+            new TransportResponseHandler<ExtensionActionResponse>() {
+
+                @Override
+                public ExtensionActionResponse read(StreamInput in) throws IOException {
+                    return new ExtensionActionResponse(in);
+                }
+
+                @Override
+                public void handleResponse(ExtensionActionResponse response) {
+                    extensionActionResponse.setResponseBytes(response.getResponseBytes());
+                    inProgressLatch.countDown();
+                }
+
+                @Override
+                public void handleException(TransportException exp) {
+                    logger.debug("Transport request failed", exp);
+                    byte[] responseBytes = ("Request failed: " + exp.getMessage()).getBytes(StandardCharsets.UTF_8);
+                    extensionActionResponse.setResponseBytes(responseBytes);
+                    inProgressLatch.countDown();
+                }
+
+                @Override
+                public String executor() {
+                    return ThreadPool.Names.GENERIC;
+                }
+            };
+        try {
+            transportService.sendRequest(
+                extension,
+                ExtensionsOrchestrator.REQUEST_EXTENSION_HANDLE_TRANSPORT_ACTION,
+                new ExtensionHandleTransportRequest(request.getAction(), request.getRequestBytes()),
+                extensionActionResponseTransportResponseHandler
+            );
+        } catch (Exception e) {
+            logger.info("Failed to send transport action to extension " + extension.getName(), e);
+        }
+        inProgressLatch.await(5, TimeUnit.SECONDS);
```

Comment:
Just trying to understand a standard time to wait here. How are we deciding it? And should we keep it same across the file
[1] https://github.com/opensearch-project/OpenSearch/blob/feature/extensions/server/src/main/java/org/opensearch/extensions/ExtensionsOrchestrator.java#L376
[2] https://github.com/opensearch-project/OpenSearch/blob/feature/extensions/server/src/main/java/org/opensearch/extensions/ExtensionsOrchestrator.java#L520

Reply:
This should probably be a global setting that a user can adjust.  I had to make the same timing assumption for the Rest Responses.

Reply:
That is a good point. I'll make a static def and use it across. 
100 feels like a lot. But i'll start with that, we can later fix it.